test/helpers.lua

```diff
@@ -87,10 +87,14 @@ helpers.find_msg_in_log_queue = function(msg, strict)
 end
 
 helpers.teardown = function(httpd)
+    local host = httpd.host
+    local port = httpd.port
     httpd:stop()
-    helpers.retrying({}, function()
-        local r = http_client.request('GET', helpers.base_uri)
-        return r == nil
+    helpers.retrying({
+        timeout = 1,
+    }, function()
+        local s, _ = socket.tcp_connect(host, port)
+        return s == nil
     end)
```

Comment:
`helpers.retrying()` retries when its function *raises* an error. For example:

```diff
diff --git a/test/helpers.lua b/test/helpers.lua
index a9a80aa..9f33068 100644
--- a/test/helpers.lua
+++ b/test/helpers.lua
@@ -94,7 +94,7 @@ helpers.teardown = function(httpd)
         timeout = 1,
     }, function()
         local s, _ = socket.tcp_connect(host, port)
-        return s == nil
+        assert(s == nil, 'assert that the http server is stopped')
     end)
 end
```

It would be even better to don't lean on GC in a frequest operation that opens a new file descriptor (to don't hit the limit):

```diff
diff --git a/test/helpers.lua b/test/helpers.lua
index a9a80aa..0ee0bf6 100644
--- a/test/helpers.lua
+++ b/test/helpers.lua
@@ -94,7 +94,10 @@ helpers.teardown = function(httpd)
         timeout = 1,
     }, function()
         local s, _ = socket.tcp_connect(host, port)
-        return s == nil
+        if s ~= nil then
+            s:close()
+        end
+        assert(s == nil, 'assert that the http server is stopped')
     end)
 end
```